core/pkg/filestream/loop_process.go

```diff
@@ -33,29 +33,36 @@ type processedChunk struct {
 }
 
 func (fs *fileStream) addProcess(task processTask) {
-	fs.processChan <- task
+	select {
+	case fs.processChan <- task:
+
+	// If the filestream dies, this prevents us from blocking forever.
+	case <-fs.deadChan:
+	}
 }
 
-func (fs *fileStream) processRecord(record *service.Record) {
+func (fs *fileStream) processRecord(record *service.Record) error {
 	switch x := record.RecordType.(type) {
 	case *service.Record_History:
-		fs.streamHistory(x.History)
+		return fs.streamHistory(x.History)
 	case *service.Record_Summary:
-		fs.streamSummary(x.Summary)
+		return fs.streamSummary(x.Summary)
 	case *service.Record_Stats:
 		fs.streamSystemMetrics(x.Stats)
+		return nil
 	case *service.Record_OutputRaw:
 		fs.streamOutputRaw(x.OutputRaw)
+		return nil
 	case *service.Record_Exit:
 		fs.streamFinish(x.Exit)
+		return nil
 	case *service.Record_Preempting:
 		fs.streamPreempting()
+		return nil
```

Comment:
why not make all the stream functions return an error (i know that currently it will be nil, but these can fail as well, so maybe useful for future changes)?

Reply:
fair point, added this for consistency